upstash_vector/core/index_operations.py

```diff
@@ -64,27 +67,45 @@ def upsert(
             ]
         )
         ```
+
+        #OR
+
+        ```python
+        from upstash_vector import Data
+        res = index.upsert(
+            vectors=[
+                Data(id="id5", data="Goodbye-World", metadata={"metadata_f": "metadata_v"}),
+                Data(id="id6", data="Hello-World"),
+            ]
+        )
+        ```
         """
+
         vectors = convert_to_vectors(vectors)
-        payload = [
-            {"id": vector.id, "vector": vector.vector, "metadata": vector.metadata}
-            for vector in vectors
-        ]
+        payload = convert_to_payload(vectors)
 
-        return self._execute_request(payload=payload, path=UPSERT_PATH)
+        path = UPSERT_PATH
+        for vector in vectors:
+            if isinstance(vector, Data):
+                path = UPSERT_DATA_PATH
+                break
+
+        return self._execute_request(payload=payload, path=path)
 
     def query(
         self,
-        vector: Union[List[float], SupportsToList],
+        vector: Optional[Union[List[float], SupportsToList]] = None,
+        data: Optional[str] = None,
```

Comment:
We need to put this argument at the end to preserve backward compatibility so that the users who invoked this method something like `index.query([0.0], 10)` wouldn't get unexpected errors